Adding message and chat HTML

Yuval-Vino opened the pull request:
### Desired Outcome

Adding chat feature to Hotails

### Implemented Changes

Added message and chat html
updated message views in order to fit the HTML files
updated navbar chat button to redirect to message HTML
updated hotails urls added message and chat htmls
updated function to daycare model - get_daycare_primary_image_url (as implemnted in #89  for showing daycare picture in chat)
also added CSS for the relevant htmls

### Connected Issue/Story

Resolves #99 


#### Changelog

- [ ] The CHANGELOG has been updated
- [X] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage
 Added test for the HTML's that checking that we are in the right site
***test_shown_only_relevant_contacts***
***test_shown_only_relevant_message_in_chat***
***test_send_new_message***
***test_cant_send_blank_message***

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [X] This PR does not require updating any documentation

##### Adding pictures of the url
### before picking specifc user to talk
![07174943-e82d-4df8-b131-7e900eb3ca22](https://user-images.githubusercontent.com/99268065/166158803-0b467086-701e-4262-85a0-d1484d19af12.jpg)

### after choosing chat (as a dogowner)
![58c7d26e-be18-4d03-9df8-bd2ae6b0eef2](https://user-images.githubusercontent.com/99268065/166158814-f1875153-f87d-45a8-b388-9d91672ee797.jpg)

### after choosing chat (as daycare)
![21730a1c-eaa7-47c9-beb3-4193a8b29a44](https://user-images.githubusercontent.com/99268065/166158823-81c9c73f-f7d7-49a5-bc4e-c986bd6faca3.jpg)

Yuval-Vino committed:
Adding message and chat HTML
author Yuval-Vino <99268065+Yuval-Vino@users.noreply.github.com>

ErezCohenn reviewed (commented):
Niceee @Yuval-Vino 
don't forget to assign yourself, and add `ready for review` label,
and the most important tests are missing

Yuval-Vino commented:
> Niceee @Yuval-Vino don't forget to assign yourself, and add `ready for review` label, and the most important tests are missing

Done! 
Added test
### this PR is ready for reviews!!

ErezCohenn commented on daycare/models.py:
```diff
@@ -50,6 +50,12 @@ def create(email, username, password, name, description, price_per_day, capacity
+            return daycare_images.first().url
+        return "../../static/images/daycare-default-profile-image.jpeg"
+
 
```
This code was implemented in #89 right? so I think it needs to be in another separate commit and add in the description on the PR that this PR depends on #89

Yuval-Vino replied:
It is indicated in the description

ErezCohenn commented on message/test_html.py:
```diff
@@ -0,0 +1,60 @@
+class TestMessageHtml:
+    def test_shown_only_relevant_contacts(self, client):
+        daycare = random.choice([daycare for daycare in DayCare.objects.all() if 'static' in daycare.user.username])
+        client.force_login(user=daycare.user)
```
overall in tests, I think you cant use random methods, so if the test fails or succeeds you cant detect what made the reason for the tests to succeed/fail, therefor its should be explicit.

Omeramsc replied:
Tests should never have a random element in them. you should test all the **different** relevant possibilities explicitly. you should be able to depend on the test results.

Yuval-Vino replied:
Thanks for the good reviews guys!
removed random, now we test all users and all their contacts

ErezCohenn reviewed (changes requested):
left some comments below, @Yuval-Vino take a look and tell me what you think

Omeramsc commented on message/templates/messages.html:
```diff
@@ -0,0 +1,48 @@
+            {% endfor %}
+
+                    </tbody>
+                </table>
```
Is there a table here? where is the opening tag?

Yuval-Vino replied:
you're right, this is leftovers from a different try. 
Removed!

Omeramsc commented on message/templates/messages.html:
```diff
@@ -0,0 +1,48 @@
+                </a>
+            {% endfor %}
+
+                    </tbody>
```
I don't see a <tbody> tag.

Yuval-Vino replied:
you're right, this is leftovers from a different try. 
Removed!

Omeramsc commented:
Please lint and re-orgenize `message/templates/messages.html`, there is a mess in the tags there (closing tags with no opening tags, and seems like divs that are not closed either? hard to read)

Omeramsc commented on message/test_html.py:
```diff
@@ -0,0 +1,60 @@
+        daycare = random.choice([daycare for daycare in DayCare.objects.all() if 'static' in daycare.user.username])
+        client.force_login(user=daycare.user)
+        response = client.get('/messages/')
+        assert response.status_code == 200
```
we already mention the status code tests are redundant...

Yuval-Vino commented:
> left some comments below, @Yuval-Vino take a look and tell me what you think

thanks for the reviews, I address them all :)

Yuval-Vino commented:
> Please lint and re-orgenize `message/templates/messages.html`, there is a mess in the tags there (closing tags with no opening tags, and seems like divs that are not closed either? hard to read)

Done! also addressed all of your comments , let me know if you see more possible changes to improve

ErezCohenn reviewed (approved):
LGTM !
don't forget to rebase.
also I think you should squash the commit `Fixing the PR after comments` to `Adding message and chat HTML`

OfirMatasas reviewed (approved):
Looks amazing! great job, @Yuval-Vino

tamirmatok reviewed (approved):
LGTM ! great job @Yuval-Vino .

ormergi reviewed (changes requested):
Hi Yuval, please squash intermediate commits

Yuval-Vino commented:
> Hi Yuval, please squash intermediate commits

Done

ormergi commented on main/templates/main/base_template.html:
```diff
@@ -27,7 +27,7 @@
                     <li class="nav-item"><a class="nav-link text-white" href="">Orders</a></li>
                     <li class="nav-item"><a class="nav-link text-white" href="">Search</a></li>
-                    <li class="nav-item"><a class="nav-link text-white" href="">Chats</a></li>
+                    <li class="nav-item"><a class="nav-link text-white" href="/messages">Chats</a></li>
```
nit: I would call it chat

Yuval-Vino replied:
name chat is occupied for chat between 2 users.
From '/messages' you can be redirected to '/chat' with a specific chat partner

ormergi reviewed (changes requested):
Hi Yuval, 
Please fix the commit message and body.
Also see my inline comments.

Yuval-Vino commented on message/test_html.py:
```diff
@@ -0,0 +1,62 @@
+
+
+@pytest.mark.django_db()
+class TestMessageHtml:
```
Done

Yuval-Vino commented on message/test_html.py:
```diff
@@ -0,0 +1,62 @@
+        for daycare in static_daycares:
+            client.force_login(user=daycare.user)
+            response = client.get('/messages/')
+            assert response.status_code == 200
```
Done

Omeramsc commented:
Please resolve conflicts ASAP if you want it in for the demo.
There is room for improvements, but It is good enough to get in.

Yuval-Vino committed:
Merge branch 'main' into chatHTML

Omeramsc commented:
+1 LGTM
